unit-tests/__main__test.py

```diff
@@ -40,3 +46,66 @@ def test_missing_style(tmp_path: Path):
         _ = perform(configuration, [tmp_path], ['missing'], [], verbose=False)
     assert ex.value.message \
            == 'Style "missing" is not defined by the configuration.'
+
+
+def test_cli_args_without_config():
+    """
+    Checks command reports an error if -configuration is not provided.
+    """
+
+    argv = ["stylist"]
+    error = io.StringIO()
+
+    with patch.object(sys, "argv", argv):
+        with raises(SystemExit) as ex:
+            with contextlib.redirect_stderr(error):
+                _ = __parse_cli()
+    assert ex.value != 0
+    assert "required: -configuration" in error.getvalue()
+
+
+def test_cli_args_with_missing_config():
+    """
+    Checks command reports an error if -configuration does not exist.
+    """
+
+    argv = ["stylist", "-configuration", "nosuch.py", "source.f90"]
+    error = io.StringIO()
+
+    with patch.object(sys, "argv", argv):
+        with raises(SystemExit) as ex:
+            with contextlib.redirect_stderr(error):
+                _ = __parse_cli()
+    assert ex.value != 0
+    assert "configuration file does not exist" in error.getvalue()
+
+
+def test_cli_args_with_text_config():
+    """
+    Checks command reports an error if -configuration is not python.
+    """
+
+    with NamedTemporaryFile(suffix=".txt") as conf:
```

Comment:
We use the PyTest framework which provides a number of magic fixtures for creating temporary directories and handling mocking. If you are not familiar with it have a look at https://docs.pytest.org/en/latest/how-to/tmp_path.html and https://docs.pytest.org/en/latest/how-to/monkeypatch.html.

It may tidy up the test code a little.